src/web/mjs/connectors/mangaz.mjs

```diff
@@ -11,36 +11,38 @@ export default class Mangaz extends Connector {
         this.url = 'https://www.mangaz.com';
     }
 
-    // async _getMangas() {
-    //     let mangaList = [];
-    //     const request = new Request(new URL('/manga/official/search', this.url), this.requestOptions);
-    //     const data = await this.fetchDOM(request, 'span:last-child > a');
-    //     const pageCount = parseInt(data[0].href.match(/(\d)+$/)[1]);
-    //     for(let page = 1; page <= pageCount; page++) {
-    //         let mangas = await this._getMangasFromPage(page);
-    //         mangaList.push(...mangas);
-    //     }
-    //     return mangaList;
-    // }
+    async _getMangas() {
+        let mangaList = [];
+        for(let page = 0, run = true; run; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
 
-    // async _getMangasFromPage(page) {
-    //     const request = new Request(new URL(`/manga/official/search?page=${page}`, this.url), this.requestOptions);
-    //     const data = await this.fetchDOM(request, 'div.official-manga-panel > a');
-    //     return data.map(element => {
-    //         return {
-    //             id: this.getRootRelativeOrAbsoluteLink(element, this.url),
-    //             title: element.querySelector('.title').textContent.replace('[R18]', '').trim()
-    //         };
-    //     });
-    // }
+    async _getMangasFromPage(page) {
+        const request = new Request(new URL('/title/addpage_renewal?query=&page='+page,this.url), {
+            method:'GET',
+            headers: {
+                'X-Requested-With': 'XMLHttpRequest'
+            }
+        });
+        let data = await this.fetchDOM(request, 'h4 > a');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                title: element.text.trim()
+            };
+        });
+    }
 
-    // async _getMangaFromURI(uri) {
-    //     const request = new Request(uri, this.requestOptions);
-    //     const data = await this.fetchDOM(request, 'div.manga-detail-description > div.title');
-    //     const id = uri.pathname + uri.search;
-    //     const title = data[0].textContent.trim();
-    //     return new Manga(this, id, title);
-    // }
+    async _getMangaFromURI(uri) {
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'li.title');
+        const id = uri.pathname + uri.search;
+        const title = data[0].textContent.trim();
+        return new Manga(this, id, title);
+    }
 
     async _getPages(chapter) {
```

Comment:
The method `_getPages` is supposed to return links to images, but in this case the method will download all images of the chapter at once and store them in memory.
This can become very slow if a chapter has dozens of images and is also memory inefficient especially for users with low end systems.

There is a Connector Handler mechanism designed for such use cases.
See: https://github.com/manga-download/hakuneko/wiki/Developer-Website-Tutorial#get-protected-image-links
Sample: [SundayWebry](https://github.com/manga-download/hakuneko/blob/master/src/web/mjs/connectors/SundayWebry.mjs#L54-#L78)

In general `_getPages` would return a list of protocol links generated in the following way:
```javascript
async _getPages(chapter) {
    return [...].map(... => {
        this._createConnectorURI({
            url: ... // link to the encrypted image data
            iv: ... // integration vector used for decrypting the image
            key: ... // key used for decrypting the image
        });
    });
}

async _handleConnectorURI(payload) {
    // download encrypted image data from payload.url
    // decrypt image data with CryptoJS AES using payload.iv and payload.key
    // return image data (mime typed buffer)
}
```